cmd/soroban-rpc/internal/methods/get_events.go

```diff
@@ -302,25 +306,48 @@ type GetEventsResponse struct {
 	LatestLedger uint32      `json:"latestLedger"`
 }
 
-type eventScanner interface {
-	Scan(eventRange events.Range, f events.ScanFunction) (uint32, error)
+type eventsRPCHandler struct {
+	dbReader          db.EventReader
+	maxLimit          uint
+	defaultLimit      uint
+	logger            *log.Entry
+	networkPassphrase string
 }
 
-type eventsRPCHandler struct {
-	scanner      eventScanner
-	maxLimit     uint
-	defaultLimit uint
+func combineContractIDs(filters []EventFilter) []string {
+	contractIDSet := make(map[string]struct{})
+
+	for _, filter := range filters {
+		for _, contractID := range filter.ContractIDs {
+			contractIDSet[contractID] = struct{}{}
+		}
+	}
+
+	contractIDs := make([]string, 0, len(contractIDSet))
+	for contractID := range contractIDSet {
+		contractIDs = append(contractIDs, contractID)
+	}
+	return contractIDs
 }
 
-func (h eventsRPCHandler) getEvents(request GetEventsRequest) (GetEventsResponse, error) {
+func (h eventsRPCHandler) getEvents(ctx context.Context, request GetEventsRequest) (GetEventsResponse, error) {
+
 	if err := request.Valid(h.maxLimit); err != nil {
```

Comment:
We can also include the same check for latestLedger <= start <= oldestLedger that we do here in `getTransactions` - https://github.com/stellar/soroban-rpc/blob/main/cmd/soroban-rpc/internal/methods/get_transactions.go#L40

Reply:
Why is it necessary? It adds more pressure to the DB. I agree that we should be consistent across both endpoints but `GetLedgerRange`  does not seem to be particularly cheap.

Reply:
I agree that GetLedgerRange is expensive call but

1. This is the current behavior of getEvents to validate range and also to be consistent with other endpoints
2. We anyway going to call `GetLedgerRange` to include latest ledger in response